src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -1200,7 +1167,10 @@ private static int FormatOctal(long value, Span<byte> destination)
             ulong remaining = (ulong)value;
             Span<byte> digits = stackalloc byte[32]; // longer than any possible octal formatting of a ulong
 
+            digits.Slice(0, digits.Length - 1).Fill(0x30); // Prefill with '0' chars except the last, which should remain 0x0 (null terminator)
```

Comment:
The `WriteRightAlignedBytesAndGetChecksum` call below takes care of this already.

Reply:
Ah, deleting this line. The tests pass still.